mass2/core/channels.py

```diff
@@ -653,3 +654,57 @@ def from_df(
                 npulses=len(df),
             )
         return Channels(channels, description)
+
+    def save_analysis(self, path: Path | str, overwrite: bool = False) -> None:
+        """Save an analysis-in-progress completely to a directory
+
+        Parameters
+        ----------
+        path : Path | str
+            Directory to save work in. If it doesn't exist, its parent should.
+        overwrite : bool, optional
+            If `path` exists, whether to overwrite it, by default False
+        """
+        if isinstance(path, str):
+            path = pathlib.Path(path)
+        if os.path.exists(path):
+            if os.path.exists(path / "data_all.pkl") and not overwrite:
+                raise ValueError(f"Set save_analysis(...overwrite=True) to overwrite the existing {path=}")
+            if not os.path.isdir(path):
+                raise ValueError(f"save_analysis(path={path}) requires that `path` be a directory, if it exists")
+        else:
+            os.mkdir(path)
+            assert os.path.exists(path) and os.path.isdir(path)
+
+        channels = {}
+        for ch_num, ch in self.channels.items():
+            fname = path / f"data_chan{ch_num:04d}.parquet"
+            df = ch.df.drop("pulse", "timestamp", "subframecount")
+            df.write_parquet(fname)
+            steps = ch.steps.trim_dead_ends(None, drop_debug=True)
+            channels[ch_num] = dataclasses.replace(ch, df=pl.DataFrame(), df_history=[], noise=None, steps=steps)
+        data = dataclasses.replace(self, channels=channels, bad_channels={})
```

Comment:
I'd be tempted to preserve the `bad_channels` info as well just for completeness. There are more questions you can answer with only the checkpoint (saved analysis) if you include bad channels.  But thats easy enough to add in later if it actually ever matters.